tests/test_models.py

```diff
@@ -2496,16 +2496,24 @@ def test_add_private_link(self):
         assert_in(link, self.project.private_links)
 
     def test_has_anonymous_link(self):
```

Comment:
This test is getting large. I understand it to be better practice with unit tests to make them small, with each test named with a full sentence describing its purpose. For example, in this case, I would prefer to see this test split in two:

```
def test_has_anonymous_user_is_true_for_private_link_anon(self):
def test_has_anonymous_user_is_false_for_private_link_not_anon(self):
```

Then I think it would make sense to add more tests, given the number of variables interacting in this feature as noted in the PR description.